mapreduce/src/main/java/org/infinispan/hadoop/mapreduce/impl/InfinispanRecordWriter.java

```diff
@@ -0,0 +1,37 @@
+package org.infinispan.hadoop.mapreduce.impl;
+
+import org.apache.hadoop.mapreduce.RecordWriter;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+import org.infinispan.client.hotrod.RemoteCache;
+import org.infinispan.hadoop.mapreduce.KeyValueConverter;
+
+import java.io.IOException;
+
+/**
+ * @param <K>
+ * @param <V>
+ * @author gustavonalle
+ * @since 0.1
+ */
+public class InfinispanRecordWriter<K, V> extends RecordWriter<K, V> {
```

Comment:
possible improvement:
we could use batch per segment to avoid interact with the remote cache for each key/value. We could find the segment for each key and when the batch is full (configurable) we could flush it to the cache (using putAll)